dipy/denoise/pca_noise_estimate.pyx

```diff
@@ -47,6 +48,9 @@ def pca_noise_estimate(data, gtab, patch_radius=1, correct_bias=True,
     smooth : int
         Radius of a Gaussian smoothing filter to apply to the noise estimate
         before returning. Default: 2.
+    image_as_samples : bool
+        Whether to use images as rows (samples) for PCA (algorithm in [1]_) or
+        to use images as columns (features). Default: True (images as samples).
```

Comment:
Following

> I don't believe that transposing matrix X is a good idea, since this then turns the voxels into features, and then the algorithms really would be quite different? While this would remove the zero-eigenvalues (for cases where they would appear before the transpose), I don't think this problem is equivalent.

in https://github.com/dipy/dipy/discussions/2678#discussioncomment-4342309, then what would be a use case for the case where the variable value is `false`? (despite https://github.com/dipy/dipy/pull/2707/files#diff-33a3d513671c2a32557d87cc9b4c1ef1c7fb068ed4a4110abb2a222cde658669R103 and having added a test)

Reply:
Allowing for `False` allows the code to work as previously (before my changes). In truth, the two approaches are sort of doing something quite similar, and it is not obvious what the best approach really is. But it is clear that `True` is the canonical algorithm.

This is for estimating the noise level, rather than for doing the denoising using that estimate noise level. I don't believe that transposing X is a good idea for the denoising step (and this certainly isn't what the literature prescribes)

(I will remove the default argument from the documentation)

Reply:
Then either the application case `True` vs. `False` should be documented (if relevant) or a warning should be raised if `False`. Also, if the default has changed, this should also go through some cycle of warning users that the default behavior has changed (regardless of the previous comment).

Reply:
I have decided to leave the non-canonical algorithm, which was the original algorithm in the code, as the default. I have documented this thoroughly now.

Reply:
:+1: for me, but probably other reviewers or people more knowledgeable should weigh in.